src/engines.jl

```diff
@@ -61,15 +64,57 @@ get_test_engine()::String = TEST_ENGINE_PROVISION.provider()
 # Release test engine. Notifies the provider that this engine is no longer in use.
 release_test_engine(name::String) = TEST_ENGINE_PROVISION.releaser(name)
 
-function set_engine_name_provider(provider::Function)
+"""
+    set_engine_name_provider(provider::Function)
+
+Set a provider for test engine names.
+
+The provider is called by each test to select an engine to run the test with. The default
+provider selects a name from a pool of available test engines.
+
+# Examples
+
+```
+set_engine_name_provider(() -> "MyEngine")
+set_engine_name_provider(() -> my_custom_engine_selector())
+
+```
+"""
+function set_engine_name_provider!(provider::Function)
     return TEST_ENGINE_PROVISION.provider = provider
 end
 
-function set_engine_name_releaser(releaser::Function)
+"""
+    set_engine_name_releaser(releaser::Function)
+
+Set a releaser for test engine names.
+
+The releaser will be called after a test has been run. The default releaser
+marks an engine in the test engine pool as available for use by another test.
+
+# Examples
+
+```
+set_engine_name_releaser((::String) -> nothing)
+set_engine_name_releaser((name::String) -> delete_engine(context, name))
```

Comment:
given the comment says just it marks it as available for use by another test, the deletion here seems out of sync with that vibe

Reply:
The deletion was to go with the provider example that created an engine - thus one new engine per test. Apart from the provider part already being removed, it's not an example that works without more context to explain it.